libsolidity/interface/CompilerStack.cpp

```diff
@@ -1045,6 +1045,16 @@ void CompilerStack::compileContract(
 		solAssert(false, "Assembly exception for deployed bytecode");
 	}
 
+	// Throw an error if EIP-170 limits are exceeded.
+	if (
+		m_evmVersion >= langutil::EVMVersion::spuriousDragon() &&
+		compiledContract.object.bytecode.size() > 0x6000
+	)
+		m_errorReporter.warning(
```

Comment:
Should we turn this into an error? If yes, what kind?

Reply:
I've seen people using Ganache with `allowUnlimitedContractSize` during development, and enabling optimizations when their contracts are ready. While this workflow seems risky to me, it apparently works. Making this an error would prevent them from using this workflow, so I think a warning is the right choice here.

Reply:
I don't see the reason to turn this into an error. We actually don't know which chain we are compiling for.

Reply:
Well EVM Version supposed to specify this, isn't it? At least in the past there was correlation.

Reply:
If you do not go all the way to code generation, you will not see an error. Then, if you request the bytecode, suddenly an errors appears. This could be weird.